homeassistant/components/smart_meter_b_route/config_flow.py

```diff
@@ -0,0 +1,123 @@
+"""Config flow for Smart Meter B Route integration."""
+
+import logging
+from typing import Any
+
+from momonga import Momonga, MomongaSkJoinFailure, MomongaSkScanFailure
+from serial import Serial
+from serial.tools.list_ports import comports
+import voluptuous as vol
+
+from homeassistant.components.usb import (
+    UsbServiceInfo,
+    get_serial_by_id,
+    human_readable_device_name,
+)
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_DEVICE, CONF_ID, CONF_PASSWORD
+from homeassistant.core import callback
+
+from .const import DOMAIN, ENTRY_TITLE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def activate_ascii_mode_if_needed(device: str) -> None:
+    """Activate ASCII mode if needed."""
+    with Serial(device, 115200) as serial:
+        serial.write(b"ROPT\r")
+        _LOGGER.debug(serial.readline())
+        mode = serial.read_until(b"\r").strip()
+        _LOGGER.info("Mode: %s", mode)
+        if mode == b"OK 00":
+            _LOGGER.info("Activating ASCII mode")
+            serial.write(b"WOPT 01\r")
+        else:
+            _LOGGER.info("ASCII mode already active")
+        serial.flush()
```

Comment:
You can still use this integration on your own though :)

Reply:
(but massive bonus points for bringing back to the community)

Reply:
Ok, will contact the creator, will be Japanese practice.

Reply:
Hum... I think I understand why the library author did not add it to its library.

In the [constructor documentation](https://www.ratoc-e2estore.com/blog/2023/06/wsuha-01), it is said (Chat GPT translated):

> To set up the RS-WSUHA for retrieving power data from a smart meter using B-route, additional firmware parameters must be configured. These settings should be manually adjusted via Teraterm or a terminal, rather than being written into the program, to avoid exceeding the write limits of the internal Flash memory. After verifying changes with Rxxx commands, avoid further Wxxx command execution unless necessary.

What should we do... Asking people to make these commands manually make the product non accessible and if we check before writing, it should be fine too...

Reply:
The library author added that processing into the library.